### ISSUE TITLE
[Bug report] requestPermissionExtend() is always denied if only the READ_MEDIA_IMAGES permission is used, and the READ_MEDIA_VIDEO permission is missing in the manifest.

### ISSUE THREAD

**Issue by thanglq1** [fluttercandies/flutter_photo_manager#1143]
### Version

3.1.1

### Platforms

Android

### Device Model

Samsung A71

### flutter info

```sh
[✓] Flutter (Channel stable, 3.22.1, on macOS 14.5 23F79 darwin-arm64, locale en-VN)
    • Flutter version 3.22.1 on channel stable at /Volumes/Private-Data/Working/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (3 weeks ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/thanglq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/thanglq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.89.1)
    • VS Code at /Users/thanglq/Desktop/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • SM A715F (mobile)               • R58NC2ZKY9T           • android-arm64  • Android 13 (API 33)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.5 23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.5 23F79 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 125.0.6422.142
    ! Error: Browsing on the local area network for iPad của Anhthu. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### How to reproduce?

My app works only with images, so I just required the READ_MEDIA_IMAGES permission in the manifest. When I called the requestPermissionExtend method, it was always denied. Therefore, I had to add the READ_MEDIA_VIDEO permission in the manifest. However, my app does not actually need this permission because it is not working with videos

**Comment by AlexV525** [comment]
Did you set a proper `RequestType`?

**Comment by thanglq1** [comment]
Hi @AlexV525. Thank you for pointing that out. I added the ```RequestType```, and it's working now

**Comment by srnden** [comment]
Hi. @AlexV525 

> Did you set a proper `RequestType`?

Does not work for api less than 29.

The "READ_MEDIA_IMAGES" permission is not enough. It is necessary to add 
```
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="32"/>
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="28"/>
```

 I don't quite understand why it's necessary to request write permissions if you only need to read.

https://github.com/fluttercandies/flutter_photo_manager/blob/d61bda8a7546b6bbd24896f14da3e2b43a3be2e9/android/src/main/kotlin/com/fluttercandies/photo_manager/permission/impl/PermissionDelegate23.kt#L26

**Comment by AlexV525** [comment]
@srnden which means you didn't even look at the README.

**Comment by srnden** [comment]
@AlexV525, I'm sorry, but I didn't see in the README why WRITE_EXTERNAL_STORAGE is required for Android 9.

https://github.com/fluttercandies/flutter_photo_manager/blob/b3f27d3ba3093c6852d3d7e4b01392b7c6ce09d7/android/src/main/kotlin/com/fluttercandies/photo_manager/core/PhotoManagerPlugin.kt#L154C9-L160C14

In version flutter_photo_manager 2.7.2, WRITE_EXTERNAL_STORAGE was optional, as far as I can see. Everything worked fine for me without specifying this permission. Maybe I don't understand something. If you have the time and desire, please explain.

**Comment by AlexV525** [comment]
If you're using an outdated version please upgrade since we've fixed related things starting from v3.1.0.

**Comment by srnden** [comment]
Hello, @AlexV525
https://youtu.be/cSFHyBvedUU

in video: 
emu api 28. 
part 1 - pm3.2.0 with WRITE_EXTERNAL_STORAGE, result - authorized (flutter 3.22.2);
part 2 - pm3.2.0 without WRITE_EXTERNAL_STORAGE, result - denied (flutter 3.22.2);
part 2 - pm2.7.2 without WRITE_EXTERNAL_STORAGE, result - authorized (flutter 3.13.9).

**Comment by AlexV525** [comment]
> Hello, @AlexV525 https://youtu.be/cSFHyBvedUU
> 
> in video: emu api 28. part 1 - pm3.2.0 with WRITE_EXTERNAL_STORAGE, result - authorized (flutter 3.22.2); part 2 - pm3.2.0 without WRITE_EXTERNAL_STORAGE, result - denied (flutter 3.22.2); part 3 - pm2.7.2 without WRITE_EXTERNAL_STORAGE, result - authorized (flutter 3.13.9).

Fixed in v3.2.2.
